agent/installer/registry.go

```diff
@@ -11,18 +11,24 @@ import (
 type osk8sInstaller interface{}
 type k8sInstallerMap map[string]osk8sInstaller
 type osk8sInstallerMap map[string]k8sInstallerMap
-type filterBundlePair struct {
+type filterOsBundleMap struct {
 	osFilter string
 	osBundle string
 }
-type filterBundleList []filterBundlePair
+type filterK8sBundleMap struct {
```

Comment:
Suggestion - if we can use only `k8sFilter`, we probably would not need a dummy `v1.22.3` version to be mapped. This makes sense for OS, because we are treating all `20.04.*` as `20.04.1`, but we cannot map the same when it comes to k8s version.

I could be wrong though - I am trying to avoid the hardcoding of `v1.22.3` in `addBundleInstaller` call.

Reply:
Ah, but of course. We don't need a map at all. A simple array with filter(s) will do just fine. Thanks, Anusha!

Reply:
You're totally right. I made the updates accordingly. However, it would be great if you have a look again because I don't feel that I came up with the best solution. I'm open for suggestions.